code/script.js

```diff
@@ -303,41 +324,91 @@ const filterCharacters = (keep) => {
+        )
+    } else {
+        alert(
+        `No, the person doesn't have ${attribute}! Remove all that have ${attribute}`
+        )
+    } 
+    } else if (category === 'hair color') {
+      if (keep) {
+        alert(
+          `Yes, the person has ${value} hair color! Keep all that have ${value} hair color`
+        )
+    } else {
+        alert(
+          `No, the person doesn't have ${value} hair color! Remove all that have ${value} hair color`
+        )
+    } 
+    } else {
+      if (keep) {
+        alert(
+          `Yes, the person has ${value} eyes! Keep all that has ${value} eyes`
+       )
     } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+        alert(
+          `No, the person doesn't have ${value} eyes! Remove all that have ${value} eyes`
+      )
     }
   }
 
   // filter to keep or remove based on the keep variable.
   /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
     or 
     charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
-
+  if (keep) {
+    charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] === currentQuestion.value)
+  } else {
+    charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] !== currentQuestion.value)
+  }
   // Invoke a function to redraw the board with the remaining people.
+  generateBoard(charactersInPlay)
 }
-
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (suspect) => {
   // store the interaction from the player in a variable.
   // remember the confirm() ?
+  const confirmGuess = confirm(`Do you want to make a guess on ${suspect}`)
   // If the player wants to guess, invoke the checkMyGuess function.
+    if (confirmGuess === true) {
+      checkMyGuess(suspect)
+  } else {     
+    }
 }
 
 // If you confirm, this function is invoked
 const checkMyGuess = (suspect) => {
+  let winOrLoseText = document.getElementById('winOrLoseText')
   // 1. Check if the suspect is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
-}
-
+  if (suspect === secret.name) {
+    // 2. Set a Message to show in the win or lose section accordingly
+    winOrLoseText.innerHTML= `Congratulations, you're guess on ${suspect} is correct!`
+  } else {
+    winOrLoseText.innerHTML= `Sorry, ${suspect} is not the right person!`
+  }
+    // 3. Show the win or lose section
+    winOrLose.style.display = "flex"
+    // 4. Hide the game board
+    board.style.display = "none"
+  }
+  
 // Invokes the start function when website is loaded
 start()
+console.log('Start function invoked: ', charactersInPlay)
+
+// Invokes the generateBoard function when website is loaded
+generateBoard()
+
```

Comment:
The generateBoard() can be removed as the invoke of start()  already has the generateBoard function inside of it.